src/ctia/stores/atom/actor.clj

```diff
@@ -1,11 +1,19 @@
 (ns ctia.stores.atom.actor
   (:require [ctia.schemas.actor :refer [NewActor StoredActor realize-actor]]
-            [ctia.stores.atom.common :as mc]))
+            [ctia.stores.atom.common :as mc]
+            [schema.core :as s]))
 
 (def swap-actor (mc/make-swap-fn realize-actor))
 
-(mc/def-create-handler handle-create-actor
-  StoredActor NewActor swap-actor (mc/random-id "actor"))
+(mc/def-create-handler-from-realized handle-create-actor StoredActor)
```

Comment:
Let's convert this to a fn generator, instead of a macro, call validate, see if we can find a way to make the validate conditional on test mode.  If not, validate at all times on input, better safe than sorry, and this function will have other entry points that realize -- consider this is also the entry point for import/replication etc...